ruby/README.md

```diff
@@ -0,0 +1,79 @@
+- Prefer `select` over `find_all`.
+- Prefer `map` over `collect`.
+- Prefer `reduce` over `inject`.
+- Prefer `&:method_name` to `{ |item| item.method_name }` for simple method
+  calls.
+- Use `_` for unused block parameters.
+- Prefix unused variables or parameters with underscore (`_`).
+- Suffix variables holding a factory with `_factory` (`user_factory`).
+- Use a leading underscore when defining instance variables for memoization.
+- Use `%()` for single-line strings containing double-quotes that require
+  interpolation.
+- Use `?` suffix for predicate methods.
+- Use `def self.method`, not `class << self`.
+- Use `def` with parentheses when there are arguments. 
+- Use heredocs for multi-line strings.
+- Order class methods above instance methods.
+- Prefer method invocation over instance variables.
+- Avoid optional parameters. Does the method do too much?
+- Avoid monkey-patching.
+- Generate necessary [Bundler binstubs] for the project, such as `rake` and
+  `rspec`, and add them to version control.
+- Prefer classes to modules when designing functionality that is shared by
+  multiple models.
+- Prefer `private` when indicating scope. Use `protected` only with comparison
+  methods like `def ==(other)`, `def <(other)`, and `def >(other)`.
+
+[bundler binstubs]: https://github.com/sstephenson/rbenv/wiki/Understanding-binstubs
+
+## Bundler
+
+- Specify the [Ruby version] to be used on the project in the `Gemfile`.
+- Use a [pessimistic version] in the `Gemfile` for gems that follow semantic
+  versioning, such as `rspec`, `factory_bot`, and `capybara`.
+- Use a [versionless] `Gemfile` declarations for gems that are safe to update
+  often, such as pg, thin, and debugger.
+- Use an [exact version] in the `Gemfile` for fragile gems, such as Rails.
+
+[ruby version]: http://bundler.io/v1.3/gemfile_ruby.html
+[exact version]: http://robots.thoughtbot.com/post/35717411108/a-healthy-bundle
+[pessimistic version]: http://robots.thoughtbot.com/post/35717411108/a-healthy-bundle
+[versionless]: http://robots.thoughtbot.com/post/35717411108/a-healthy-bundle
+
+## Ruby Gems
+
+- Declare dependencies in the `<PROJECT_NAME>.gemspec` file.
+- Reference the `gemspec` in the `Gemfile`.
+- Use [Appraisal] to test the gem against multiple versions of gem dependencies
+  (such as Rails in a Rails engine).
+- Use [Bundler] to manage the gem's dependencies.
+- Use continuous integration (CI) to show build status within the code review
+  process and to test against multiple Ruby versions.
+
+[appraisal]: https://github.com/thoughtbot/appraisal
+[bundler]: http://bundler.io
+
+## Ruby JSON and GraphQL APIs
+
+- Review the recommended practices outlined in Heroku's [HTTP API Design Guide]
+  before designing a new API.
+- Write integration tests for your API endpoints. When the primary consumer of
+  the API is a JavaScript client maintained within the same code base as the
+  provider of the API, write [feature specs]. Otherwise write [request specs].
```

Comment:
Seems like its more common for use to do both does the team agree with this portion of the entry?

Reply:
Yeah good call, we typically aim for both.

Reply:
As a follow up, I personally prefer both just by the way I tend to write new entries in our GraphQL API. For example, I'll consider the setup of the field or mutation and what I need back from it for the remainder of my story and slice that off as an individual PR. This would require a request spec as there is nothing else using it until further PRs where I implement the FE